tasks/modules.py

```diff
@@ -0,0 +1,57 @@
+import os
+
+
+class GoModule:
+    """A Go module abstraction."""
+
+    def __init__(self, path, targets=None, condition=lambda: True, dependencies=None):
+        self.path = path
+        self.targets = targets if targets else ["."]
+        self.dependencies = dependencies if dependencies else []
+        self.condition = condition
+
+    def __version(self, agent_version):
+        """Return the module version for a given Agent version.
+        >>> mods = [GoModule("."), GoModule("./pkg/util/log")]
+        >>> [mod.__version("7.27.0") for mod in mods]
+        ["v7.27.0", "v0.27.0"]
+        """
+        if self.path == ".":
+            return "v" + agent_version
+
+        return "v0" + agent_version[1:]
+
+    def tag(self, agent_version):
+        """Return the module tag name for a given Agent version.
+        >>> mods = [GoModule("."), GoModule("pkg/util/log")]
+        >>> [mod.tag("7.27.0") for mod in mods]
+        [["6.27.0", "7.27.0"], ["pkg/util/log/v0.27.0"]]
+        """
+        if self.path == ".":
+            return ["6" + agent_version[1:], "7" + agent_version[1:]]
```

Comment:
We always return Agent 6 and Agent 7 tags. Do we want to have an option for releases that are just for one of them?

Reply:
We can leave it like this for now. We can change it when we decouple the release cycles of A6 and A7 in the future.